error: Expected 1 arguments, got 0
  --> tests/ui/invalid_proto_pymethods.rs:20:8
   |
20 |     fn truediv_expects_one_argument(&self) -> PyResult<()> {
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Expected 1 arguments, got 0
  --> tests/ui/invalid_proto_pymethods.rs:28:8
   |
28 |     fn truediv_expects_one_argument_py(&self, _py: Python<'_>) -> PyResult<()> {
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `signature` cannot be used with magic method `__bool__`
  --> tests/ui/invalid_proto_pymethods.rs:39:31
   |
39 |     #[pyo3(name = "__bool__", signature = ())]
   |                               ^^^^^^^^^

error: `text_signature` cannot be used with magic method `__bool__`
  --> tests/ui/invalid_proto_pymethods.rs:47:31
   |
47 |     #[pyo3(name = "__bool__", text_signature = "")]
   |                               ^^^^^^^^^^^^^^

error[E0034]: multiple applicable items in scope
  --> tests/ui/invalid_proto_pymethods.rs:56:1
   |
56 | #[pymethods]
   | ^^^^^^^^^^^^ multiple `__pymethod___richcmp____` found
   |
note: candidate #1 is defined in an impl for the type `EqAndRichcmp`
  --> tests/ui/invalid_proto_pymethods.rs:56:1
   |
56 | #[pymethods]
   | ^^^^^^^^^^^^
note: candidate #2 is defined in an impl for the type `EqAndRichcmp`
  --> tests/ui/invalid_proto_pymethods.rs:56:1
   |
56 | #[pymethods]
   | ^^^^^^^^^^^^
   = note: this error originates in the macro `::pyo3::impl_::pyclass::generate_pyclass_richcompare_slot` which comes from the expansion of the attribute macro `pymethods` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0592]: duplicate definitions with name `__pymethod___richcmp____`
  --> tests/ui/invalid_proto_pymethods.rs:56:1
   |
56 | #[pymethods]
   | ^^^^^^^^^^^^
   | |
   | duplicate definitions for `__pymethod___richcmp____`
   | other definition for `__pymethod___richcmp____`
   |
   = note: this error originates in the macro `::pyo3::impl_::pyclass::generate_pyclass_richcompare_slot` which comes from the expansion of the attribute macro `pymethods` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0034]: multiple applicable items in scope
  --> tests/ui/invalid_proto_pymethods.rs:56:1
   |
56 | #[pymethods]
   | ^^^^^^^^^^^^ multiple `__pymethod___richcmp____` found
   |
note: candidate #1 is defined in an impl for the type `EqAndRichcmp`
  --> tests/ui/invalid_proto_pymethods.rs:56:1
   |
56 | #[pymethods]
   | ^^^^^^^^^^^^
note: candidate #2 is defined in an impl for the type `EqAndRichcmp`
  --> tests/ui/invalid_proto_pymethods.rs:56:1
   |
56 | #[pymethods]
   | ^^^^^^^^^^^^
   = note: this error originates in the attribute macro `pymethods` (in Nightly builds, run with -Z macro-backtrace for more info)
